### PR TITLE
attempting to split up dialog.spec.ts

### PR DESCRIPTION
**Pull request by dan100110**

### CODE REVIEW

File: node/test/sensitive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,1396 @@
...
+  let allScenarios: DialogScenario[] = await findAllScenarios();
+  const allLessons: Lesson[] = await findAllLessons();
+
+  const lessonById: Record<string, Lesson> = {};
+  allLessons.forEach((lesson) => {
+    lessonById[lesson.lessonId] = lesson;
+  });
+
+  beforeEach(async () => {
+    if (!MOCKING_DISABLED) {
```
Comment (beatthat) [inline node/test/sensitive.spec.ts:61]: there's no reason to do this MOCKING_DISABLED stuff at all anymore

File: node/test/helpers.ts [hunk 2]
```diff
@@ -8,10 +8,13 @@ import yn from 'yn';
 import axios from 'axios';
 import { Express } from 'express';
 import request from 'supertest';
+import MockAxios from 'axios-mock-adapter';
 
 export const MOCKING_DISABLED = yn(process.env.MOCKING_DISABLED || 0);
 export const DIALOG_ENDPOINT = process.env.DIALOG_ENDPOINT || '/dialog';
 
+export const MOCK_AXIOS = new MockAxios(axios);
```
Comment (beatthat) [inline node/test/helpers.ts:16]: seems weird and possibly dangerous to do this at the static/module level as opposed to inside the scope of a test as it has been up until now...do we really need this?

File: node/test/sensitive.spec.ts [hunk 3]
```diff
@@ -0,0 +1,1396 @@
...
+    //       .map((m) => (m.data as TextData).text)
+    //     ).to.include.oneOf(["Would you like to proceed?", "Are you comfortable continuing with this lesson?"])
+    //   expect(
+    //     (responseStartSession.body.response as OpenTutorResponse[])
+    //       .filter((m) => m.type === ResponseType.MainQuestion)
+    //     ).to.be.empty;
+    // });
+
+    // Update the session data for testing dialog behavior with streaks of negative answers
+    let lessonIdq6 = 'q6';
```
Comment (beatthat) [inline node/test/sensitive.spec.ts:520]: why would this not be `const`?

File: node/test/survey-says.spec.ts [hunk 4]
```diff
@@ -0,0 +1,1134 @@
...
+  let allScenarios: DialogScenario[] = await findAllScenarios();
+  const allLessons: Lesson[] = await findAllLessons();
+
+  const lessonById: Record<string, Lesson> = {};
+  allLessons.forEach((lesson) => {
+    lessonById[lesson.lessonId] = lesson;
+  });
+
+  beforeEach(async () => {
+    if (!MOCKING_DISABLED) {
```
Comment (beatthat) [inline node/test/survey-says.spec.ts:56]: mocking cannot be disabled, remove check

**Review by beatthat — CHANGES_REQUESTED** [review]
Look ok to me overall.

Would be better to figure out a leaner way to generate the sessionhistory data in the tests, but i can see why you would want to defer that for another time.

Definitely you need to get rid of the MOCKING_DISABLED stuff (mocking cannot be disabled)

Also curious if we really need to move the `MockAxios` instantiation out to a static-module const? We never needed this before, and as a rule I would say it's not a great idea. If you really need it, then go with it

File: node/test/helpers.ts (hunk 2 above)
Comment (dan100110) [reply node/test/helpers.ts:16]: when I tried to use separate mockAxios instances, only one file would properly return the mock responses.  I couldn't figure out why it was doing that, so my current solution was to use only one instance.  If you can figure out what's going on there please let me know.

**Review by beatthat — APPROVED** [review]

**Pull request merged**